### PR TITLE
Migrate istio importer #1059

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: vulnerabilities/importers/istio.py [hunk 1]
```diff
@@ -6,165 +6,128 @@
...
+        release_date = None
+        if published_date:
+            release_date = parser.parse(published_date).replace(tzinfo=pytz.UTC)
+
+        constraints = []
+
+        for release in data.get("releases") or []:
+            # If it is of form "All releases prior to x"
+            if "All releases prior" in release:
+                release = release.strip()
```
Comment (TG1999) [inline vulnerabilities/importers/istio.py:62]:
```suggestion
                _ ,_, release = release.strip().rpartition(" ")
```

File: vulnerabilities/importers/istio.py [hunk 2]
```diff
@@ -6,165 +6,128 @@
...
+                release = release.split(" ")
```
Comment (TG1999) [inline vulnerabilities/importers/istio.py:63]:
```suggestion
```

File: vulnerabilities/importers/istio.py [hunk 3]
```diff
@@ -6,165 +6,128 @@
...
+                constraints.append(
+                    VersionConstraint(version=SemverVersion(release[4]), comparator="<")
```
Comment (TG1999) [inline vulnerabilities/importers/istio.py:65]:
```suggestion
                    VersionConstraint(version=SemverVersion(release), comparator="<")
```

File: vulnerabilities/importers/istio.py [hunk 4]
```diff
@@ -6,165 +6,128 @@
...
-                    affected_packages=affected_packages,
-                    references=[
-                        Reference(
-                            reference_id=data["title"],
-                            url=f"https://istio.io/latest/news/security/{data['title']}/",
-                        )
-                    ],
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(type="golang", name="istio"),
```
Comment (TG1999) [inline vulnerabilities/importers/istio.py:100]:
```suggestion
                    package=PackageURL(type="golang", namespace="istio.io", 
name="istio"),
```

File: vulnerabilities/importers/istio.py [hunk 5]
```diff
@@ -6,165 +6,128 @@
...
+                    affected_version_range=GolangVersionRange(constraints=constraints),
+                )
+            )
+
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(type="github", name="istio"),
```
Comment (TG1999) [inline vulnerabilities/importers/istio.py:107]:
```suggestion
                    package=PackageURL(type="github", namespace="istio",  name="istio"),
```

File: vulnerabilities/tests/test_data/istio/istio-expected.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  {
+    "aliases": [
+      "CVE-2019-12243"
+    ],
+    "summary": "Incorrect access control.",
+    "affected_packages": [
+      {
+        "package": {
+          "type": "golang",
+          "namespace": null,
```
Comment (TG1999) [inline vulnerabilities/tests/test_data/istio/istio-expected.json:11]:
```suggestion
          "namespace": "istio.io",
```

**Review by TG1999 — COMMENTED** [review]
Suggestions:

**Review by pombredanne — APPROVED** [review]
LGTM with a few namespace adjustments

**Pull request merged**